File: 45-checkboxes/java/Checkboxes.java
import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;

import java.util.List;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.is;

public class Checkboxes {
    WebDriver driver;

    @Before
    public void setUp() throws Exception {
        driver = new FirefoxDriver();
    }

    @Test
    public void checkboxDiscoveryTest() {
        driver.get("http://the-internet.herokuapp.com/checkboxes");
        List<WebElement> checkboxes = driver.findElements(By.cssSelector("input[type=\"checkbox\"]"));

        System.out.println("With .attribute('checked')");
        for (WebElement checkbox : checkboxes) {
            System.out.println(String.valueOf(checkbox.getAttribute("checked")));
        }

        System.out.println("\nWith .selected?");
        for (WebElement checkbox : checkboxes) {
            System.out.println(checkbox.isSelected());
        }
    }

    @Test
    public void checkboxOption1Test() throws Exception {
        // check if 1st checkbox is not checked
        driver.get("http://the-internet.herokuapp.com/checkboxes");
        WebElement checkbox1 = driver.findElement(By.xpath("/html/body/div[2]/div/div/form/input[1]"));
        assertThat(checkbox1.isSelected(), is(false));
    }

    @Test
    public void checkboxOption2Test() throws Exception {
        // check if 2nd checkbox is checked
        driver.get("http://the-internet.herokuapp.com/checkboxes");
        driver.get("http://the-internet.herokuapp.com/checkboxes");
        WebElement checkbox1 = driver.findElement(By.xpath("/html/body/div[2]/div/div/form/input[2]"));
        assertThat(checkbox1.isSelected(), is(true));
    }

    @After
    public void tearDown() throws Exception {
        driver.quit();
    }
}

File: 45-checkboxes/python/checkboxes.py
# -*- coding: utf-8 -*-
"""
Implementation of http://elementalselenium.com/tips/45-checkboxes
"""

import unittest
from selenium import webdriver


class Checkboxes(unittest.TestCase):

    def setUp(self):
        self.driver = webdriver.Firefox()

    def tearDown(self):
        self.driver.quit()

    def test_list_values_for_different_approaches(self):
        driver = self.driver
        driver.get('http://the-internet.herokuapp.com/checkboxes')
        checkboxes = driver.find_elements_by_css_selector('input[type="checkbox"]')

        print("With .get_attribute('checked')")
        for checkbox in checkboxes:
            print(checkbox.get_attribute('checked'))

        print("\nWith .is_selected")
        for checkbox in checkboxes:
            print(checkbox.is_selected())

        assert checkboxes[-1].get_attribute('checked')
        assert checkboxes[-1].is_selected()
        assert checkboxes[0].get_attribute('checked')
        assert checkboxes[0].is_selected()

if __name__ == "__main__":
    unittest.main()
